### PR TITLE
add rgbled_ncp5623c driver

### PR DESCRIPTION
**Pull request by CUAVcaijie**
**Test data / coverage**
compile
debugging

**Describe problem solved by the proposed pull request**
Add a new rgbled(ncp5623c) driver

**Describe possible alternatives**
None

**Additional context**
None

### CODE REVIEW

**Comment by LorenzMeier** [comment]
Style check failed - please run `make format`:
http://ci.px4.io:8080/blue/organizations/jenkins/PX4%2FFirmware/detail/PR-10889/1/pipeline/

**Comment by klopezal** [comment]
Could you also add the driver to fmu-v4pro? Thanks!

**Comment by CUAVcaijie** [comment]
> Could you also add the driver to fmu-v4pro? Thanks!

I will compile for v4pro tomorrow. But I don't have v4pro hardware.

File: src/drivers/rgbled_ncp5623c/CMakeLists.txt [hunk 1]
```diff
@@ -0,0 +1,43 @@
+############################################################################
+#
+#   Copyright (c) 2015 PX4 Development Team. All rights reserved.
```
Comment (dagar) [inline src/drivers/rgbled_ncp5623c/CMakeLists.txt]: year

File: src/drivers/rgbled_ncp5623c/rgbled_ncp5623c.cpp [hunk 2]
```diff
@@ -0,0 +1,423 @@
+/****************************************************************************
+ *
+ *   Copyright (c) 2012-2015 PX4 Development Team. All rights reserved.
```
Comment (dagar) [inline src/drivers/rgbled_ncp5623c/rgbled_ncp5623c.cpp]: year

File: src/drivers/rgbled_ncp5623c/rgbled_ncp5623c_params.c [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+/**
+ * RGB Led brightness limit
+ *
+ * Set to 0 to disable, 1 for minimum brightness up to 31 (max)
+ *
+ * @min 0
+ * @max 31
+ * @group System
+ */
+PARAM_DEFINE_INT32(NCP5623C_MAXBRT, 31);
```
Comment (dagar) [inline src/drivers/rgbled_ncp5623c/rgbled_ncp5623c_params.c]: Should we make this a generic system wide parameter? I'm guessing people don't need to adjust the brightness of different devices separately.

File: src/drivers/rgbled_ncp5623c/rgbled_ncp5623c_params.c [hunk 4]
```diff
@@ -0,0 +1,55 @@
+/****************************************************************************
+ *
+ *   Copyright (c) 2015 PX4 Development Team. All rights reserved.
```
Comment (dagar) [inline src/drivers/rgbled_ncp5623c/rgbled_ncp5623c_params.c]:
```suggestion
 *   Copyright (c) 2018 PX4 Development Team. All rights reserved.
```

File: src/drivers/rgbled_ncp5623c/rgbled_ncp5623c.cpp (hunk 2 above)
Comment (dagar) [inline src/drivers/rgbled_ncp5623c/rgbled_ncp5623c.cpp]:
```suggestion
 *   Copyright (c) 2018 PX4 Development Team. All rights reserved.
```

File: src/drivers/rgbled_ncp5623c/CMakeLists.txt (hunk 1 above)
Comment (dagar) [inline src/drivers/rgbled_ncp5623c/CMakeLists.txt]:
```suggestion
#   Copyright (c) 2018 PX4 Development Team. All rights reserved.
```

**Comment by jinger26** [comment]
@davids5 FYI

File: src/drivers/rgbled_ncp5623c/rgbled_ncp5623c_params.c (hunk 3 above)
Comment (mrpollo) [reply src/drivers/rgbled_ncp5623c/rgbled_ncp5623c_params.c]: I think we should, what does this mean for our first-time contributor?

**Comment by mrpollo** [comment]
We have some comments regarding your changes mostly around formatting, please let us know if you need assistance, you can find us on [Slack](https://dev.px4.io/en/#support).

First-time contributor @CUAVcaijie 🎉  Thank you!

File: src/drivers/rgbled_ncp5623c/rgbled_ncp5623c_params.c (hunk 3 above)
Comment (davids5) [reply src/drivers/rgbled_ncp5623c/rgbled_ncp5623c_params.c]: I believe this is HW specific setting for this chip i.e. a register setting and should be not is the  @ group System - @dagar - what groups should be used?

File: src/drivers/rgbled_ncp5623c/rgbled_ncp5623c.cpp [hunk 5]
```diff
@@ -0,0 +1,423 @@
...
+#include <systemlib/err.h>
+
+#include <board_config.h>
+
+#include <drivers/drv_led.h>
+#include <lib/led/led.h>
+
+#include "uORB/topics/parameter_update.h"
+
+#define ADDR				 0x39	    /**< I2C adress of NCP5623C */
```
Comment (davids5) [inline src/drivers/rgbled_ncp5623c/rgbled_ncp5623c.cpp]: @dagar - OK on address - Address looks to be unique in PX4 code base

File: src/drivers/rgbled_ncp5623c/rgbled_ncp5623c.cpp [hunk 6]
```diff
@@ -0,0 +1,423 @@
...
+	work_queue(LPWORK, &_work, (worker_t)&RGBLED_NPC5623C::led_trampoline, this, 0);
+	return OK;
+}
+
+int
+RGBLED_NPC5623C::probe()
+{
+	_retries = 4;
+
+	return write(NCP5623_LED_CURRENT, 0x00);
```
Comment (davids5) [inline src/drivers/rgbled_ncp5623c/rgbled_ncp5623c.cpp]: Just a comment:  probe usually tries to read a `whoami` code from a register. The chip does not have one: sigh! 

perhaps NCP5623_LED_OFF should be used here. (we may change this after testing)

File: src/drivers/rgbled_ncp5623c/rgbled_ncp5623c.cpp [hunk 7]
```diff
@@ -0,0 +1,423 @@
...
+	msg[4] = NCP5623_LED_PWM1 | (uint8_t(_g * _brightness) & 0x1f);
+	msg[6] = NCP5623_LED_PWM2 | (uint8_t(_b * _brightness) & 0x1f);
+
+	return transfer(&msg[0], 7, nullptr, 0);
+	return 0;
+}
+
+
+void
+RGBLED_NPC5623C::update_params()
```
Comment (davids5) [inline src/drivers/rgbled_ncp5623c/rgbled_ncp5623c.cpp]: The TCA driver has. 
```
/**
 * RGB Led brightness limit
 *
 * Set to 0 to disable, 1 for minimum brightness up to 15 (max)
 *
 * @min 0
 * @max 15
 * @group System
 */
PARAM_DEFINE_INT32(LED_RGB_MAXBRT, 15);
```
This driver uses 0 - 31, but converts 0 to 1. @dagar shulde we unify this?
I wonder if this should be 0-100%

**Comment by davids5** [comment]
@CUAVcaijie - Thank you again. 

@dagar This has been tested and works well. 0 load changes, start and stop work correctly. Led tracks unboard led. Please merge.

**Comment by dagar** [comment]
Looks good to merge. We can revisit this later to properly handle multiple instances, as well as how that extends to parameters.

**Comment by dagar** [comment]
Thanks @CUAVcaijie!

**Pull request merged**